apps/air-discount-scheme/backend/src/app/modules/user/user.service.ts

```diff
@@ -16,27 +16,35 @@ import {
   EinstaklingarGetForsjaRequest,
 } from '@island.is/clients/national-registry-v2'
 import environment from '../../../environments/environment'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
 
 @Injectable()
 export class UserService {
   constructor(
     private readonly flightService: FlightService,
     private readonly nationalRegistryService: NationalRegistryService,
     private readonly nationalRegistryIndividualsApi: EinstaklingarApi,
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
   ) {}
 
-  async getRelations(authUser: AuthUser): Promise<string[]> {
-    const relations = await this.nationalRegistryIndividualsApi
-      .withMiddleware(
-        new AuthMiddleware(
-          authUser,
-          environment.nationalRegistry
-            .authMiddlewareOptions as AuthMiddlewareOptions,
-        ),
-      )
-      .einstaklingarGetForsja(<EinstaklingarGetForsjaRequest>{
-        id: authUser.nationalId,
-      })
+  async getRelations(authUser: AuthUser): Promise<Array<string>> {
+    let relations: string[] = []
+    try {
+      relations = await this.nationalRegistryIndividualsApi
+        .withMiddleware(
+          new AuthMiddleware(
+            authUser,
+            environment.nationalRegistry
+              .authMiddlewareOptions as AuthMiddlewareOptions,
+          ),
+        )
+        .einstaklingarGetForsja(<EinstaklingarGetForsjaRequest>{
+          id: authUser.nationalId,
+        })
+    } catch (e) {
+      this.logger.error(e)
```

Comment:
In my previous reviews I was missing the fact that this endpoint should still return the authenticated user, even though the forsja call fails. Now I understand why you are handling this error at all.

The general point is that errors should always be handled as high up as possible, either by business logic, or by the user. Arguably it should not be in this function but somewhere higher, but it doesn't really matter in this case since the lines between client logic and domain/business logic is a bit blurry here.

Reply:
I find this very reasonable.
I guess initially I went overboard with logging errors after trying to debug staging on release - The fact that there were so few traces astounded me.